components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpUtils.java

```diff
@@ -35,6 +35,19 @@
 
     private FtpUtils() {
     }
+    
+    public static String extractDirNameFromAbsolutePath(String path) {
+        // default is unix so try with '/'
+        // otherwise force File.separator
+        if (path.endsWith("/") || path.endsWith(File.separator)) {
+            path = path.substring(0, path.length() - 1);
+        }
+        int startIndex = path.lastIndexOf(File.separator);
+        if (startIndex == -1) {
+            startIndex = path.lastIndexOf("/");
+        }
+        return path.substring(startIndex + 1, path.length());
```

Comment:
If there is no slash then start index is -1 and then I think we should make it more clear that we dont substring it then, eg

if (startIndex != -1) {
   ..
} else {
   // no directory then return file as-is
   return path;
}

Reply:
Also there is a stripPath method from FileUtil you can use for the last part: org.apache.camel.util.FileUtil#stripPath